### PR TITLE
remove outdated handoff event

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: wei.liu <wei.liu@zilliz.com>
Related to #20190

### CODE REVIEW

File: internal/querycoordv2/observers/handoff_observer.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func (ob *HandoffObserver) schedule(ctx context.Context) {
 
 		case resp, ok := <-watchChan:
 			if !ok {
-				log.Error("watch Segment handoff loop failed because watch channel is closed!")
+				panic("watch Segment handoff loop failed because watch channel is closed!")
```
Comment (jiaoew1991) [inline internal/querycoordv2/observers/handoff_observer.go:178]: should we panic here?

Comment (weiliu1031) [reply internal/querycoordv2/observers/handoff_observer.go]: Fix it!
usually when we lost connection with etcd, the service role will exit, panic here is meaningless.

File: internal/querycoordv2/observers/handoff_observer.go [hunk 2]
```diff
@@ -131,15 +131,15 @@ func (ob *HandoffObserver) reloadFromStore(ctx context.Context) error {
 			log.Error("reloadFromKV: unmarshal failed", zap.Any("error", err.Error()))
```
Comment (congqixia) [inline internal/querycoordv2/observers/handoff_observer.go:131]: NIT: `zap.Error(err)`

File: internal/querycoordv2/observers/handoff_observer.go [hunk 3]
```diff
@@ -116,30 +115,31 @@ func (ob *HandoffObserver) StartHandoff(collectionIDs ...int64) {
...
 	for _, value := range handoffReqValues {
 		segmentInfo := &querypb.SegmentInfo{}
 		err := proto.Unmarshal([]byte(value), segmentInfo)
 		if err != nil {
-			log.Error("reloadFromKV: unmarshal failed", zap.Any("error", err.Error()))
+			log.Error("reloadFromKV: unmarshal failed", zap.Error(err))
 			return err
 		}
-		ob.tryHandoff(ctx, segmentInfo)
+		ob.cleanEvent(ctx, segmentInfo)
```
Comment (yah01) [inline internal/querycoordv2/observers/handoff_observer.go:134]: return error

**Pull request merged**